apps/avifdec.c

```diff
@@ -176,21 +180,38 @@ int main(int argc, char * argv[])
                 return 1;
             }
             imageDimensionLimit = (uint32_t)value;
+        } else if (arg[0] == '-') {
+            fprintf(stderr, "ERROR: unrecognized flag %s\n\n", arg);
+            syntax();
+            return 1;
         } else {
             // Positional argument
             if (!inputFilename) {
                 inputFilename = arg;
             } else if (!outputFilename) {
                 outputFilename = arg;
             } else {
-                fprintf(stderr, "Too many positional arguments: %s\n", arg);
+                fprintf(stderr, "Too many positional arguments: %s\n\n", arg);
                 syntax();
                 return 1;
             }
         }
 
         ++argIndex;
     }
+    // Parse additional positional arguments if any
+    while (argIndex < argc) {
+        const char * arg = argv[argIndex];
+        if (!inputFilename) {
+            inputFilename = arg;
+        } else if (!outputFilename) {
+            outputFilename = arg;
+        } else {
+            fprintf(stderr, "Too many positional arguments: %s\n\n", arg);
+            syntax();
+            return 1;
+        }
+    }
```

Comment:
Consider moving this loop to line 85, because it is the only place where it is used.